src/Table.js

```diff
@@ -697,27 +709,88 @@ export class Table {
     /*
         Convert items into a map of items by model type
     */
-    groupByType(items, params={}) {
-        let result = {}
-        for (let item of items) {
-            let type = item[this.typeField] || '_unknown'
-            let list = result[type] = result[type] || []
-            let model = this.schema.models[type]
-            let preparedItem
+    groupByType(items, params = {}) {
+        return groupByMapping(items, item => item[this.typeField] || '_unknown', (item, type) => {
+            const model = this.schema.models[type]
             if (typeof params.hidden === 'boolean' && !params.hidden) {
-                let fields = model.block.fields
-                preparedItem = {}
-                for (let [name, field] of Object.entries(fields)) {
+                const preparedItem = {}
+                const fields = model.block.fields
+                for (const [name, field] of Object.entries(fields)) {
                     if (!(field.hidden && params.hidden !== true)) {
                         preparedItem[name] = item[name]
                     }
                 }
-            } else {
-                preparedItem = item
+                return preparedItem
             }
-            list.push(preparedItem)
-        }
-        return result
+            return item
+        })
+    }
+
+    /**
+     * this is a function passed to the DataLoader that given an array of Get commands
+     * will group all of them by TableName and make all Keys unique and then will perform
+     * a BatchGet request and process the response of the BatchRequest to return an array
+     * with the corresponding response in the same order as it was requested.
+     *
+     * @param commands
+     * @returns {Promise<*>}
+     */
+    async groupByTableAndBatchGet(commands) {
```

Comment:
this is the method that DataLoader will use to batch get all items, it receives an array of Get commands and return an array of `{ Item }` so the get command continues thinking it was just a regular Get